### ISSUE TITLE
FNX3-16993 ⁃ (GeckoView) IllegalStateException Display already acquired

### ISSUE THREAD

**Issue by jonalmeida** [mozilla-mobile/fenix#341]
Sentry: https://sentry.prod.mozaws.net/operations/fenix/issues/5176161/?query=is:unresolved

```
java.lang.IllegalStateException: Display already acquired
    at org.mozilla.geckoview.GeckoSession.acquireDisplay(GeckoSession.java:1800)
    at org.mozilla.geckoview.GeckoView.setSession(GeckoView.java:342)
    at org.mozilla.geckoview.GeckoView.setSession(GeckoView.java:305)
    at mozilla.components.browser.engine.gecko.GeckoEngineView.render(GeckoEngineView.kt:51)
    at mozilla.components.feature.session.EngineViewPresenter.renderSession$feature_session_release(EngineViewPresenter.kt:50)
    at mozilla.components.feature.session.EngineViewPresenter.start(EngineViewPresenter.kt:30)
    at mozilla.components.feature.session.SessionFeature.start(SessionFeature.kt:26)
    at java.lang.reflect.Method.invoke(Method.java)
    at androidx.lifecycle.ClassesInfoCache$MethodReference.invokeCallback(ClassesInfoCache.java:215)
    at androidx.lifecycle.ClassesInfoCache$CallbackInfo.invokeMethodsForEvent(ClassesInfoCache.java:193)
    at androidx.lifecycle.ClassesInfoCache$CallbackInfo.invokeCallbacks(ClassesInfoCache.java:184)
    at androidx.lifecycle.ReflectiveGenericLifecycleObserver.onStateChanged(ReflectiveGenericLifecycleObserver.java:36)
    at androidx.lifecycle.LifecycleRegistry$ObserverWithState.dispatchEvent(LifecycleRegistry.java:355)
    at androidx.lifecycle.LifecycleRegistry.forwardPass(LifecycleRegistry.java:293)
    at androidx.lifecycle.LifecycleRegistry.sync(LifecycleRegistry.java:333)
    at androidx.lifecycle.LifecycleRegistry.moveToState(LifecycleRegistry.java:138)
    at androidx.lifecycle.LifecycleRegistry.handleLifecycleEvent(LifecycleRegistry.java:124)
    at androidx.fragment.app.Fragment.performStart(Fragment.java:2485)
    at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManager.java:1494)
    at androidx.fragment.app.FragmentManagerImpl.addAddedFragments(FragmentManager.java:2646)
    at androidx.fragment.app.FragmentManagerImpl.executeOpsTogether(FragmentManager.java:2416)
    at androidx.fragment.app.FragmentManagerImpl.removeRedundantOperationsAndExecute(FragmentManager.java:2372)
```

This is also being tracked and investigated on: https://github.com/mozilla-mobile/reference-browser/issues/534

Full stackstace:
<details>

```
java.lang.IllegalStateException: Display already acquired
    at org.mozilla.geckoview.GeckoSession.acquireDisplay(GeckoSession.java:1800)
    at org.mozilla.geckoview.GeckoView.setSession(GeckoView.java:342)
    at org.mozilla.geckoview.GeckoView.setSession(GeckoView.java:305)
    at mozilla.components.browser.engine.gecko.GeckoEngineView.render(GeckoEngineView.kt:51)
    at mozilla.components.feature.session.EngineViewPresenter.renderSession$feature_session_release(EngineViewPresenter.kt:50)
    at mozilla.components.feature.session.EngineViewPresenter.start(EngineViewPresenter.kt:30)
    at mozilla.components.feature.session.SessionFeature.start(SessionFeature.kt:26)
    at java.lang.reflect.Method.invoke(Method.java)
    at androidx.lifecycle.ClassesInfoCache$MethodReference.invokeCallback(ClassesInfoCache.java:215)
    at androidx.lifecycle.ClassesInfoCache$CallbackInfo.invokeMethodsForEvent(ClassesInfoCache.java:193)
    at androidx.lifecycle.ClassesInfoCache$CallbackInfo.invokeCallbacks(ClassesInfoCache.java:184)
    at androidx.lifecycle.ReflectiveGenericLifecycleObserver.onStateChanged(ReflectiveGenericLifecycleObserver.java:36)
    at androidx.lifecycle.LifecycleRegistry$ObserverWithState.dispatchEvent(LifecycleRegistry.java:355)
    at androidx.lifecycle.LifecycleRegistry.forwardPass(LifecycleRegistry.java:293)
    at androidx.lifecycle.LifecycleRegistry.sync(LifecycleRegistry.java:333)
    at androidx.lifecycle.LifecycleRegistry.moveToState(LifecycleRegistry.java:138)
    at androidx.lifecycle.LifecycleRegistry.handleLifecycleEvent(LifecycleRegistry.java:124)
    at androidx.fragment.app.Fragment.performStart(Fragment.java:2485)
    at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManager.java:1494)
    at androidx.fragment.app.FragmentManagerImpl.addAddedFragments(FragmentManager.java:2646)
    at androidx.fragment.app.FragmentManagerImpl.executeOpsTogether(FragmentManager.java:2416)
    at androidx.fragment.app.FragmentManagerImpl.removeRedundantOperationsAndExecute(FragmentManager.java:2372)
    at androidx.fragment.app.FragmentManagerImpl.popBackStackImmediate(FragmentManager.java:884)
    at androidx.fragment.app.FragmentManagerImpl.popBackStackImmediate(FragmentManager.java:827)
    at androidx.fragment.app.FragmentManagerImpl.popBackStackImmediate(FragmentManager.java:874)
    at androidx.fragment.app.FragmentManagerImpl.popBackStackImmediate(FragmentManager.java:827)
    at androidx.fragment.app.FragmentActivity.onBackPressed(FragmentActivity.java:190)
    at android.app.Activity.onKeyUp(Activity.java:3082)
    at android.view.KeyEvent.dispatch(KeyEvent.java:2716)
    at android.app.Activity.dispatchKeyEvent(Activity.java:3365)
    at androidx.core.app.ComponentActivity.superDispatchKeyEvent(ComponentActivity.java:108)
    at androidx.core.view.KeyEventDispatcher.dispatchKeyEvent(KeyEventDispatcher.java:84)
    at androidx.core.app.ComponentActivity.dispatchKeyEvent(ComponentActivity.java:126)
    at androidx.appcompat.app.AppCompatActivity.dispatchKeyEvent(AppCompatActivity.java:535)
    at androidx.appcompat.view.WindowCallbackWrapper.dispatchKeyEvent(WindowCallbackWrapper.java:59)
    at androidx.appcompat.app.AppCompatDelegateImpl$AppCompatWindowCallback.dispatchKeyEvent(AppCompatDelegateImpl.java:2533)
    at com.android.internal.policy.DecorView.dispatchKeyEvent(DecorView.java:342)
    at android.view.ViewRootImpl$ViewPostImeInputStage.processKeyEvent(ViewRootImpl.java:5037)
    at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:4905)
    at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4426)
    at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4479)
    at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4445)
    at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:4585)
    at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4453)
    at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:4642)
    at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4426)
    at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4479)
    at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4445)
    at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4453)
    at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4426)
    at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4479)
    at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4445)
    at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:4618)
    at android.view.ViewRootImpl$ImeInputStage.onFinishedInputEvent(ViewRootImpl.java:4779)
    at android.view.inputmethod.InputMethodManager$PendingEvent.run(InputMethodManager.java:2571)
    at android.view.inputmethod.InputMethodManager.invokeFinishedInputEventCallback(InputMethodManager.java:2081)
    at android.view.inputmethod.InputMethodManager.finishedInputEvent(InputMethodManager.java:2072)
    at android.view.inputmethod.InputMethodManager$ImeInputEventSender.onInputEventFinished(InputMethodManager.java:2548)
    at android.view.InputEventSender.dispatchInputEventFinished(InputEventSender.java:141)
    at android.os.MessageQueue.nativePollOnce(MessageQueue.java)
    at android.os.MessageQueue.next(MessageQueue.java:326)
    at android.os.Looper.loop(Looper.java:160)
    at android.app.ActivityThread.main(ActivityThread.java:6718)
    at java.lang.reflect.Method.invoke(Method.java)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)

java.lang.RuntimeException: Failed to call observer method
    at androidx.lifecycle.ClassesInfoCache$MethodReference.invokeCallback(ClassesInfoCache.java:225)
    at androidx.lifecycle.ClassesInfoCache$CallbackInfo.invokeMethodsForEvent(ClassesInfoCache.java:193)
    at androidx.lifecycle.ClassesInfoCache$CallbackInfo.invokeCallbacks(ClassesInfoCache.java:184)
    at androidx.lifecycle.ReflectiveGenericLifecycleObserver.onStateChanged(ReflectiveGenericLifecycleObserver.java:36)
    at androidx.lifecycle.LifecycleRegistry$ObserverWithState.dispatchEvent(LifecycleRegistry.java:355)
    at androidx.lifecycle.LifecycleRegistry.forwardPass(LifecycleRegistry.java:293)
    at androidx.lifecycle.LifecycleRegistry.sync(LifecycleRegistry.java:333)
    at androidx.lifecycle.LifecycleRegistry.moveToState(LifecycleRegistry.java:138)
    at androidx.lifecycle.LifecycleRegistry.handleLifecycleEvent(LifecycleRegistry.java:124)
    at androidx.fragment.app.Fragment.performStart(Fragment.java:2485)
    at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManager.java:1494)
    at androidx.fragment.app.FragmentManagerImpl.addAddedFragments(FragmentManager.java:2646)
    at androidx.fragment.app.FragmentManagerImpl.executeOpsTogether(FragmentManager.java:2416)
    at androidx.fragment.app.FragmentManagerImpl.removeRedundantOperationsAndExecute(FragmentManager.java:2372)
    at androidx.fragment.app.FragmentManagerImpl.popBackStackImmediate(FragmentManager.java:884)
    at androidx.fragment.app.FragmentManagerImpl.popBackStackImmediate(FragmentManager.java:827)
    at androidx.fragment.app.FragmentManagerImpl.popBackStackImmediate(FragmentManager.java:874)
    at androidx.fragment.app.FragmentManagerImpl.popBackStackImmediate(FragmentManager.java:827)
    at androidx.fragment.app.FragmentActivity.onBackPressed(FragmentActivity.java:190)
    at android.app.Activity.onKeyUp(Activity.java:3082)
    at android.view.KeyEvent.dispatch(KeyEvent.java:2716)
    at android.app.Activity.dispatchKeyEvent(Activity.java:3365)
    at androidx.core.app.ComponentActivity.superDispatchKeyEvent(ComponentActivity.java:108)
    at androidx.core.view.KeyEventDispatcher.dispatchKeyEvent(KeyEventDispatcher.java:84)
    at androidx.core.app.ComponentActivity.dispatchKeyEvent(ComponentActivity.java:126)
    at androidx.appcompat.app.AppCompatActivity.dispatchKeyEvent(AppCompatActivity.java:535)
    at androidx.appcompat.view.WindowCallbackWrapper.dispatchKeyEvent(WindowCallbackWrapper.java:59)
    at androidx.appcompat.app.AppCompatDelegateImpl$AppCompatWindowCallback.dispatchKeyEvent(AppCompatDelegateImpl.java:2533)
    at com.android.internal.policy.DecorView.dispatchKeyEvent(DecorView.java:342)
    at android.view.ViewRootImpl$ViewPostImeInputStage.processKeyEvent(ViewRootImpl.java:5037)
    at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:4905)
    at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4426)
    at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4479)
    at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4445)
    at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:4585)
    at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4453)
    at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:4642)
    at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4426)
    at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4479)
    at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4445)
    at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4453)
    at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4426)
    at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4479)
    at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4445)
    at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:4618)
    at android.view.ViewRootImpl$ImeInputStage.onFinishedInputEvent(ViewRootImpl.java:4779)
    at android.view.inputmethod.InputMethodManager$PendingEvent.run(InputMethodManager.java:2571)
    at android.view.inputmethod.InputMethodManager.invokeFinishedInputEventCallback(InputMethodManager.java:2081)
    at android.view.inputmethod.InputMethodManager.finishedInputEvent(InputMethodManager.java:2072)
    at android.view.inputmethod.InputMethodManager$ImeInputEventSender.onInputEventFinished(InputMethodManager.java:2548)
    at android.view.InputEventSender.dispatchInputEventFinished(InputEventSender.java:141)
    at android.os.MessageQueue.nativePollOnce(MessageQueue.java)
    at android.os.MessageQueue.next(MessageQueue.java:326)
    at android.os.Looper.loop(Looper.java:160)
    at android.app.ActivityThread.main(ActivityThread.java:6718)
    at java.lang.reflect.Method.invoke(Method.java)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```

</details>

**Comment by pocmo** [comment]
Added the a-c label. From the stracktrace that looks like something in our component. We saw that bug in the reference browser too.

**Comment by boek** [comment]
@jonalmeida Fixed in #374 also filed up a follow up in a-c here: https://github.com/mozilla-mobile/android-components/issues/1943
